src/cloud_agents/workflow/executor/step/direct.py

```
@@ -621,13 +681,13 @@ def _parse_output(content: Any, output_schema: dict[str, Any] | None) -> dict[st
 
     if output_schema:
         try:
-            return json.loads(content)
+            return json.loads(_strip_markdown_fence(content))
         except json.JSONDecodeError:
             raise ValueError(
                 f"LLM returned non-JSON response but output_schema was requested: {content[:200]}"
             )
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)